image/openssl/src/lib.rs

```diff
@@ -125,3 +148,517 @@ fn from_hw_format(value: &[u32; ECC384_SCALAR_WORD_SIZE]) -> [u8; ECC384_SCALAR_
+    let treetype = lookup_lms_algorithm_type(priv_key.tree_type).unwrap();
+    let otstype = lookup_lmots_algorithm_type(priv_key.otstype).unwrap();
+    let Ok((_, height)) = Lms::default().get_lms_parameters(&treetype) else {
+        return Err(anyhow!("Error parsing lms parameters"));
+   };
+    let mut pub_key = Some(ImageLmsPublicKey::default());
+    if let Some(x) = pub_key.as_mut() {
+        x.otstype = priv_key.otstype;
+        x.id = priv_key.id;
+        x.tree_type = priv_key.tree_type;
+    }
+    generate_lms_pubkey_helper(
+        &priv_key.id,
+        otstype,
+        height,
+        &priv_key.seed,
+        None,
+        &mut pub_key,
+        &mut None,
+    );
+    Ok(pub_key.unwrap())
+}
+
+fn sign_with_lms_key(
+    priv_key: &ImageLmsPrivKey,
+    message: &[u8],
+    nonce: &[u8],
+    lms_alg_type: LmsAlgorithmType,
+    ots_alg_type: LmotsAlgorithmType,
+    q: u32,
+) -> anyhow::Result<ImageLmsSignature> {
+    let Ok((_, height)) = Lms::default().get_lms_parameters(&lms_alg_type) else {
+         return Err(anyhow!("Error parsing lms parameters"));
+    };
+    if q >= (1 << height) {
+        return Err(anyhow!("Invalid q"));
+    }
+    let ots_sig = generate_ots_signature_helper(
+        message,
+        ots_alg_type,
+        &priv_key.id,
+        &priv_key.seed,
+        nonce,
+        q,
+    );
+    let mut sig = Some(ImageLmsSignature::default());
+    if let Some(x) = sig.as_mut() {
+        x.q = q;
+        x.ots_sig = ots_sig;
+    }
+
+    generate_lms_pubkey_helper(
+        &priv_key.id,
+        ots_alg_type,
+        height,
+        &priv_key.seed,
+        Some(q),
+        &mut None,
+        &mut sig,
+    );
+    Ok(sig.unwrap())
+}
+
+#[test]
+fn test_print_lms_private_pub_key() {
+    let mut priv_key: ImageLmsPrivKey = ImageLmsPrivKey {
+        tree_type: LmsAlgorithmType::LmsSha256N24H15 as u32,
+        otstype: LmotsAlgorithmType::LmotsSha256N24W4 as u32,
+        ..Default::default()
+    };
+    for i in 0..4 {
+        rand_bytes(&mut priv_key.id).unwrap();
+        rand_bytes(&mut priv_key.seed).unwrap();
+        let pub_key = generate_lms_pubkey(&priv_key);
+        println!("pub const VENDOR_LMS_KEY{i}_PRIVATE: ImageLmsPrivKey = {priv_key:#04x?};");
+        println!("pub const VENDOR_LMS_KEY{i}_PUBLIC: ImageLmsPublicKey = {pub_key:#04x?};");
+    }
+    for i in 0..1 {
+        rand_bytes(&mut priv_key.id).unwrap();
+        rand_bytes(&mut priv_key.seed).unwrap();
+        let pub_key = generate_lms_pubkey(&priv_key);
+        println!("pub const OWNER_LMS_KEY{i}_PRIVATE: ImageLmsPrivKey = {priv_key:#04x?};");
+        println!("pub const OWNER_LMS_KEY{i}_PUBLIC: ImageLmsPublicKey = {pub_key:#04x?};");
+    }
+}
+
+#[test]
+fn test_lms() {
+    let priv_key = ImageLmsPrivKey {
+        tree_type: LmsAlgorithmType::LmsSha256N24H5 as u32,
+        otstype: LmotsAlgorithmType::LmotsSha256N24W8 as u32,
```

Comment:
Could you add test cases using H15 and W4 since those are the parameters we are using?